### PR TITLE
Move to openscm conventions

### PR DESCRIPTION
**Pull request by znicholls**
# Pull request

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added (where applicable)
- [x] Example added (either to an existing notebook or as a new notebook, where applicable)
- [x] Description in ``CHANGELOG.rst`` added
- [x]  check that old API still returns a `dict` (moved to #161)
- [x]  new style returns new style Dataframe (moved to #161)

## Adding to CHANGELOG.rst

Please add a single line in the changelog notes similar to one of the following:

```
- (`#XX <http://link-to-pr.com>`_) Added feature which does something
- (`#XX <http://link-to-pr.com>`_) Fixed bug identified in (`#XX <http://link-to-issue.com>`_)
```

### CODE REVIEW

**Comment by rgieseke** [comment]
Column headers convention:
Lowercase, underscore, follow Pyam

Variable and acronym naming convention:
use (and enhance) ghgrenamer (https://github.com/openclimatedata/ghgrenamer), "Emissions|N2O|MAGICC AFOLU"

'B' becomes" MAGICC AFOLU"
'I' becomes "MAGICC Fossil and Industrial"

Regional split:
World|Northern Hemisphere|Land
World|R5...
(copy Pyam)

**Comment by znicholls** [comment]
> Regional split:
> World|Northern Hemisphere|Land

'World' rather than 'Global' here is jarring but it's better to be consistent with Pyam than precious

**Comment by znicholls** [comment]
from earlier discussions:

- move to long format only (in preparation for using pyam capability)

**Comment by znicholls** [comment]
@rgieseke `Atmospheric Concentrations|CO2` or `Concentrations|CO2`, I think the argument for the former is in case we one day introduce an ocean pH module to MAGICC and need `Ocean concentrations` or similar

**Comment by znicholls** [comment]
@rgieseke this one is pretty massive. Tried to keep it just to the minimum I needed, unfortunately that's quite a lot...

File: CHANGELOG.rst [hunk 1]
```diff
@@ -4,6 +4,8 @@ Changelog
 master
 ------
 
+- (`#154 <https://github.com/openclimatedata/pymagicc/pull/154>`_) Change to using openscm variables for all user facing data as well as preparing to move to using OpenSM dataframes
```
Comment (rgieseke) [inline CHANGELOG.rst]: Should be OpenSCM throughout
```suggestion
- (`#154 <https://github.com/openclimatedata/pymagicc/pull/154>`_) Change to using OpenSCM variables for all user facing data as well as preparing to move to using OpenSM dataframes
```

File: pymagicc/io.py [hunk 2]
```diff
@@ -415,8 +472,54 @@ def _read_magicc6_style_header(self, stream, metadata):
 
         return column_headers, metadata
 
+    def _get_column_headers_update_metadata(self, stream, metadata):
```
Comment (rgieseke) [inline pymagicc/io.py]: What does this do? Get headers or update metadata?

File: docs/index.rst [hunk 3]
```diff
@@ -29,6 +29,8 @@ Pymagicc
     pymagicc.api
     pymagicc.io
     pymagicc.definitions
+    pymagicc.config
+    pymagicc.utils
```
Comment (rgieseke) [inline docs/index.rst]:
```suggestion
```
Remove module with only private functions

Comment (rgieseke) [reply docs/index.rst]: It claims that you could add a commit with this directly ... can see a button to do this?

Comment (rgieseke) [reply docs/index.rst]: Or alternatively one could add something general to `pymagicc.utils.rst`

File: pymagicc/io.py (hunk 2 above)
Comment (znicholls) [reply pymagicc/io.py]: both haha

**Pull request merged**